### PR TITLE
Add Okabe-Ito color palette

### PR DESCRIPTION
**Pull request by bwiernik**
And clean up a few typos in color scale documentation.

### CODE REVIEW

File: R/scale_color_okabeito.R [hunk 1]
```diff
@@ -0,0 +1,193 @@
...
+#' @details This function is usually not called directly, but from within
+#'   [`scale_color_material()`][scale_color_material].
+#'
+#' @export
+palette_okabeito <- function(palette = "full", reverse = FALSE, order = 1:9, ...) {
+  pal <- okabeito_palettes[[palette]]
+
+  stopifnot(
+    "`order` must be a vector of integers." = is.numeric(order),
+    "All elements of `order` must be greater than 0 and less than 9." = all(order > 0 & order <= 9)
```
Comment (bwiernik) [inline R/scale_color_okabeito.R:172]:
```suggestion
    "All elements of `order` must be greater than 0 and less than 10." = all(order > 0 & order <= 9)
```

**Review by strengejacke — APPROVED** [review]

**Comment by strengejacke** [comment]
``` r
library(ggplot2)
library(see)

ggplot(iris, aes(x = Species, y = Sepal.Length, fill = Species)) +
  geom_violin() +
  theme_modern() +
  scale_fill_oi(palette = "bla")
#> Warning: Palette name not available. `palette` must be one of `full` or
#>   `black_first`.
#>   Using default palette now.
```



``` r

ggplot(iris, aes(x = Species, y = Sepal.Length, fill = Species)) +
  geom_violin() +
  theme_modern() +
  scale_fill_oi(palette = "black_first")
```



<sup>Created on 2022-06-24 by the [reprex package](https://reprex.tidyverse.org) (v2.0.1)</sup>

**Comment by IndrajeetPatil** [comment]
Really cool! Big fan of this palette since I read Wilke's dataviz book.

Thanks!

**Comment by bwiernik** [comment]
We should add the `aesthetics` argument to all of our color/full scale functions

**Comment by strengejacke** [comment]
Why do we need that?

**Comment by bwiernik** [comment]
It lets the same scale be used for multiple aesthetics, like both `"color"` and `"fill"`, and it means that we don't have to add special functions for `"edge_color"`, etc. that *ggraph* and other packages have made. All of the *ggplot2* color/fill scales have this now.

**Pull request merged**